src/screens/Home/AssetsSection.js

```diff
@@ -0,0 +1,150 @@
+    if (!balance || !categories[category]) return null;
+
+    const formattedBalance = formatFiatValue(balance?.balanceInFiat ?? 0, fiatCurrency);
+
+    const initialBalance = balance.changeInFiat ? balance.balanceInFiat.minus(balance.changeInFiat) : null;
+    const formattedChange = formatFiatChangeExtended(balance.changeInFiat, initialBalance, fiatCurrency);
+    const { title, iconName } = categories[category];
+
+    return (
+      <HomeListItem
+        title={title}
+        iconName={iconName}
+        onPress={() => navigation.navigate(ASSETS, { category })}
+        value={formattedBalance}
+        secondaryValue={formattedChange}
+        secondaryValueColor={balance.changeInFiat?.gte(0) ? colors.positive : colors.secondaryText}
+      />
+    );
+  };
+
+  const renderChainItems = (chainInfo: ?ChainInfo) => {
+    if (!chainInfo) return null;
+
+    const formattedCollectibles = chainInfo.collectibles?.toFixed() ?? '0';
+    const formattedContacts = chainInfo.contacts?.toFixed() ?? '0';
+
+    return (
+      <>
+        {renderBalanceItem(ASSET_CATEGORIES.WALLET, chainInfo.wallet)}
+        {renderBalanceItem(ASSET_CATEGORIES.DEPOSITS, chainInfo.deposits)}
+        {renderBalanceItem(ASSET_CATEGORIES.INVESTMENTS, chainInfo.investments)}
+        {renderBalanceItem(ASSET_CATEGORIES.LIQUIDITY_POOLS, chainInfo.liquidityPools)}
+        {renderBalanceItem(ASSET_CATEGORIES.REWARDS, chainInfo.rewards)}
+        {renderBalanceItem(ASSET_CATEGORIES.DATASETS, chainInfo.datasets)}
+
+        {chainInfo.collectibles != null && (
+          <HomeListItem
+            title={tRoot('assetCategories.collectibles')}
+            iconName="wallet"
+            onPress={() => navigation.navigate(ASSETS)}
+            value={formattedCollectibles}
+          />
+        )}
+
+        {chainInfo.contacts != null && (
+          <HomeListItem
+            title={t('contacts')}
+            iconName="wallet"
+            onPress={() => navigation.navigate(CONTACTS_FLOW)}
+            value={formattedContacts}
+          />
+        )}
+
+        {/* Temporary entry until other UI provided */}
+        <HomeListItem title={t('services')} iconName="wallet" onPress={() => navigation.navigate(SERVICES_FLOW)} />
+      </>
+    );
+  };
+
+  const renderChain = (chain: Chain, chainInfo: ?ChainInfo) => {
+    if (!chainInfo) return null;
+
+    const { title, iconSource, color } = chains[chain];
+    return (
+      <>
+        <HomeListHeader title={title} iconSource={iconSource} color={color} />
+        {renderChainItems(chainInfo)}
+      </>
+    );
+  };
+
+  if (!showSideChains) {
+    return <Container>{renderChainItems(wallet.ethereum)}</Container>;
+  }
+
+  return (
+    <Container>
+      {renderChain(CHAINS.ETHEREUM, wallet.ethereum)}
+      {renderChain(CHAINS.BINANCE, wallet.binance)}
+      {renderChain(CHAINS.XDAI, wallet.xdai)}
+    </Container>
+  );
+}
+
+export default AssetsSection;
+
+const Container = styled.View``;
```

Comment:
I wonder if this wouldn't cause confusion over having styled components defined level above component in current codebase. What are your thoughts?

Reply:
When reading the existing components, that put styled components first, I've found it hard to find the logic and essence of the actual components when the component size is large (say 200-300+ LOCs). By putting the main component & it's props first, it should be easier to quickly figure out it's external API while leaving the implementation details at the end of the file.

So basically the structure I suggest for new component files is:

1. Import
2. Props & related type defs
3. Main component
4. Helper functions (if there's a lot of them, put them in `utils.js` file or something)
5. Styled components &styles (if there's a lot of them & their reusable, put them in `components.js` file or something)

@poocart wdyt?